aten/src/ATen/functorch/BatchRulesLinearAlgebra.cpp

```diff
@@ -485,6 +485,49 @@ pinv_batch_rule(
     const std::optional<int64_t> rtol_bdim, bool hermitian) {
   return atol_rtol_tensor_batch_rule(ATEN_FN2(linalg_pinv, atol_rtol_tensor), input, input_bdim, atol, atol_bdim, rtol, rtol_bdim, hermitian, "linalg.pinv");
 }
+
+fourOutputs _scaled_dot_product_efficient_attention_batch_rule(
+  const Tensor& query, optional<int64_t> query_bdim,
+  const Tensor& key, optional<int64_t> key_bdim,
+  const Tensor& value, optional<int64_t> value_bdim,
+  const std::optional<Tensor>& attn_bias, optional<int64_t> attn_bias_bdim,
+  bool compute_log_sumexp,
+  double dropout_p,
+  bool is_causal,
+  c10::optional<double> scale
+) {
+  auto query_ = moveBatchDimToFront(query, query_bdim);
+  auto key_ = moveBatchDimToFront(key, key_bdim);
+  auto value_ = moveBatchDimToFront(value, value_bdim);
+  int64_t extra_batch_size = 1;
+  if (query_bdim.has_value()) {
+    extra_batch_size = query.size(0);
+  } else if (key_bdim.has_value()) {
+    extra_batch_size = key.size(0);
+  } else if (value_bdim.has_value()) {
+    extra_batch_size = value.size(0);
+  }
```

Comment:
Shouldn't here be `query_`, `key_` and `value_`?

```suggestion
  if (query_bdim.has_value()) {
    extra_batch_size = query_.size(0);
  } else if (key_bdim.has_value()) {
    extra_batch_size = key_.size(0);
  } else if (value_bdim.has_value()) {
    extra_batch_size = value_.size(0);
  }
```

Otherwise, code like this will fail:

```python
import torch
import torch.nn.functional as F

def fn(query, key, value):
    return F.scaled_dot_product_attention(query, key, value)

B = 4
query = torch.rand(32, B, 8, 128, 64, dtype=torch.float16, device="cuda")
key = torch.rand(B, 32, 8, 128, 64, dtype=torch.float16, device="cuda")
value = torch.rand(32, 8, 128, 64, dtype=torch.float16, device="cuda")
# y = F.scaled_dot_product_attention(query,key,value)
y = torch.vmap(fn, in_dims=(1, 0, None))(query, key, value)
print(y.shape)
```

Reply:
Good catch!